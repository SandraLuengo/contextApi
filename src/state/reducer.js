import chinesCities from '../fixtures/cities-of-china.json';

const initialState = {
  cities: chinesCities.cities.map(city => ({ ...city, isSelected: false })),
  searching: '',
  isAllChecked: false,
};


const reducer = (state, action) => {
  switch (action.type) {
    case 'SET_CITIES':
      return {
        ...state,
        cities: action.newValue,
      };
    case 'SET_SEARCH':
      return {
        ...state,
        searching: action.newValue,
      };
    case 'SET_IS_ALL_CHECKED':
      return {
        ...state,
        isAllChecked: action.newValue,
      };
    default:
      return state;
  }
};

export { initialState, reducer };

