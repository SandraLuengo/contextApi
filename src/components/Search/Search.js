import React from 'react';
import { useGlobalState } from '../../state/useGlobalState';
import { citiesAreChecked } from '../functions';

const Search = ({ className }) => {
  const [{ cities }, dispatch] = useGlobalState();

  const handleChange = ({ target: { value } }) => {
    dispatch({ type: 'SET_IS_ALL_CHECKED', newValue: false });
    dispatch({ type: 'SET_SEARCH', newValue: value });
    citiesAreChecked(cities, value, dispatch);
  };

  return (
    <div className={className}>
      <div className="search">
        <i className="fas fa-search search__icon" />
        <input
          className="search__input"
          placeholder="Search by name"
          onChange={e => handleChange(e)}
        />
      </div>
    </div>
  );
};

export default Search;
