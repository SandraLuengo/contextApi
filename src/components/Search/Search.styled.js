import styled from 'styled-components';
import Search from './Search';
import * as style from '../../static/styles/stylesVariables';

export default styled(Search).attrs({})`
  width: 100%;
  height: 8vh;
  display: flex;
  justify-content: center;
  align-items: center;
  border-style: none none solid none;
  border-color: #e9e9e9;
  border-width: 1.3px;
  .search {
    width: 96%;
    height: 60%;
    display: flex;
    justify-content: space-between;
    align-items: center;
    border-bottom: solid 1.2px ${style.lightGrey};
    &__icon {
      color: ${style.lightGrey};
    }
    &__input {
      width: 95%;
      border: none;
      outline: none;
      font-size: 1rem;
      &::placeholder {
        font-size: 1rem;
        color: ${style.lightGrey};
      }
    }
  }
`;
