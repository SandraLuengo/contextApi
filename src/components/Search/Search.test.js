import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { mount } from 'enzyme';
import Search from './Search';
import { Provider } from '../../state/useGlobalState';
import { initialState, reducer } from '../../state/reducer';

Enzyme.configure({ adapter: new Adapter() });

const props = { className: 'test' };

it('Search component renders correctly', () => {
  const tree = renderer
    .create(<Provider reducer={reducer} initialState={initialState}>
      <Search {...props} />
    </Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('<Search/>', () => {
  it('should exist search input', () => {
    const wrapper = mount(<Provider reducer={reducer} initialState={initialState}>
      <Search />
    </Provider>);
    expect(wrapper.find('.search__input').length).toBe(1);
  });

  it('should change search value', () => {
    const wrapper = mount(<Provider reducer={reducer} initialState={initialState}>
      <Search />
    </Provider>);
    wrapper
      .find('.search__input')
      .simulate('change', { target: { value: 'hong kong' } });
  });
});
