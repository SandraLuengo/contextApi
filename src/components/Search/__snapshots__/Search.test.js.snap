// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders correctly 1`] = `
<div
  className="test"
>
  <div
    className="search"
  >
    <i
      className="fas fa-search search__icon"
    />
    <input
      className="search__input"
      onChange={[Function]}
      placeholder="Search by name"
    />
  </div>
</div>
`;
